### ISSUE TITLE
Second window does not use the same highlighting settings as the first

### ISSUE THREAD

**Issue by hholst80** [erf/vis-highlight#2]
I assume that this is due to a misunderstanding on my part that I do not have the configuration properly setup.  The problem is obvious from a picture:

![image](https://github.com/erf/vis-highlight/assets/6200749/9d1adda2-7d76-4860-813d-c6a6471254b3)

My config:

```
-- load standard vis module, providing parts of the Lua API
require('vis')
local plug = require('plugins/vis-plug')

vis.events.subscribe(vis.events.INIT, function()
	-- Your global configuration options
	-- vis.hi.patterns['\t'] = { style = 'back:#ffffff' }
	plug.plugins.hi.patterns[' +\n'] = { style = 'back:#ff0000' }
	plug.plugins.hi.patterns['\t+'] = { style = 'fore:#303030' }
	plug.plugins.hi.patterns['\n'] = { style = 'fore:#303030' }
end)

vis.events.subscribe(vis.events.WIN_OPEN, function(win)
	-- Your per window configuration options e.g.
	-- vis:command('set number')
	vis:command('set show-tabs on')
	vis:command('set show-newlines on')
end)

-- configure plugins in an array of tables with git urls and options
local plugins = {

	-- load a plugin given a repo (https://github.com/ can be omitted and expects a 'init.lua' file)
	-- { 'erf/vis-cursors' },

	-- first parameter is a shorthand for 'url'
	{ url = 'erf/vis-cursors' },

	-- specify an alias to later use to access plugin variables (see example below)
	{ 'erf/vis-highlight', alias = 'hi' },

	-- configure themes by setting 'theme = true'. The theme 'file' will be set on INIT
	{ 'samlwood/vis-gruvbox', theme = true, file = 'gruvbox' },

	{ 'seifferth/vis-editorconfig' },
}

plug.init(plugins, true)
```

**Comment by erf** [comment]
Haven't really looked into this in a while, and not quite sure what is going on here, could you make a minimal reproduce able example / config, with a step-by-step on how to recreate it ? Thanks

**Comment by hholst80** [comment]
The minimal config (still using your pkg manager). The repro can be to open fstab file twice, the second file will show in a second window.



```lua
-- load standard vis module, providing parts of the Lua API
require('vis')
local plug = require('plugins/vis-plug')

vis.events.subscribe(vis.events.WIN_OPEN, function(win)
	vis:command('set show-newlines on')
end)

-- configure plugins in an array of tables with git urls and options
local plugins = {

	-- specify an alias to later use to access plugin variables (see example below)
	{ 'erf/vis-highlight', alias = 'hi' },
}

plug.init(plugins, true)

plug.plugins.hi.patterns[' +\n'] = { style = 'back:#ff0000' }
plug.plugins.hi.patterns['\t+'] = { style = 'fore:#303030' }
plug.plugins.hi.patterns['\n'] = { style = 'fore:#303030' }
```
![image](https://github.com/erf/vis-highlight/assets/6200749/13a8499d-7fb2-4ba9-abee-67b486456dd1)


```
#
# /etc/fstab
# Created by anaconda on Wed May 17 16:34:58 2023
#
# Accessible filesystems, by reference, are maintained under '/dev/disk/'.
# See man pages fstab(5), findfs(8), mount(8) and/or blkid(8) for more info.
#
# After editing this file, run 'systemctl daemon-reload' to update systemd
# units generated from this file.
#
/dev/mapper/vg--zink-root /                       ext4    defaults        1 1
UUID=e84be499-d22a-43b8-8b8c-9342f828b9f7 /boot                   ext4    defaults        1 2
UUID=9ED4-B0F5          /boot/efi               vfat    umask=0077,shortname=winnt 0 2
/dev/mapper/luks-2adbdb26-98ec-4759-813b-1e164fd1630a /home                   ext4    defaults,x-systemd.device-timeout=0 1 2
/dev/mapper/luks-7fd9f6e9-64c0-4e96-b9a9-e07b38993a7e /var/lib/docker         xfs     defaults,x-systemd.device-timeout=0 0 0
```

**Comment by erf** [comment]
I sort of makes out what's going on here, but please make a simpler example file and also a step by step on how to recreate as requested, and describe the expected result and the actual result. That's how to make a good issue. 

Not sure when i'm able to take a look at this so feel free to take a stab. Also the `vis-sneak` plugin is doing something similar and might hint to a solution.

**Comment by erf** [comment]
Closed via 8ff698bccd5543f48f078298e4003ef005f23561
